src/components/UserActions/UserActivity.ts

```diff
@@ -231,178 +262,195 @@ export class UserActivity extends Component {
+        const sessionHeader = document.createElement('div');
+        sessionHeader.classList.add('coveo-session-header');
+        sessionHeader.innerText = `Session ${formatDate(session.timestamp)}`;
+        return sessionHeader;
+    }
+
+    private buildSessionContent(actions: UserAction[], withFolded: boolean): HTMLLIElement[] {
+        let actionsHTML = [];
+        let actionsToDisplay = actions;
+        if (withFolded && this.options.ticketCreationDateTime && !this.hasExpandedActions) {
+            actionsToDisplay = actionsToDisplay.filter((action) => action.timestamp <= this.options.ticketCreationDateTime);
+            if (actionsToDisplay.length < actions.length) {
+                actionsHTML.push(this.buildFoldedActions());
+            }
+        }
+        actionsHTML = actionsHTML.concat(actionsToDisplay.map((action) => this.buildActionListItem(action)));
+        return actionsHTML;
+    }
+
+    private buildFoldedActions(): HTMLLIElement {
         const li = document.createElement('li');
-        li.classList.add(CLICK_EVENT_CLASS);
+        li.classList.add(FOLDED_ACTIONS_CLASS);
 
-        const dataElement = document.createElement('a');
-        dataElement.classList.add(DATA_CLASS);
-        dataElement.innerText = (action.document && action.document.title) || '';
-        dataElement.href = (action.document && action.document.clickUri) || '';
+        const span = document.createElement('span');
+        span.classList.add(TEXT_CLASS);
+        span.innerText = 'More actions';
 
-        document.createAttributeNS('svg', 'svg');
+        li.addEventListener('click', () => {
+            this.hasExpandedActions = true;
+            this.render();
+        });
 
-        li.appendChild(this.buildTitleSection(action));
-        if (action.document) {
-            li.appendChild(dataElement);
+        for (let i = 0; i < 3; i++) {
+            const el = this.buildIcon(dot);
+            li.appendChild(el);
         }
-        li.appendChild(this.buildOriginElement(action));
-        li.appendChild(this.buildIcon(duplicate));
 
+        li.appendChild(span);
+        return li;
+    }
+
+    private buildActionListItem(action: UserAction): HTMLLIElement {
+        let li: HTMLLIElement;
+
+        switch (action.type) {
+            case UserActionType.Click:
+                li = this.buildClickEvent(action);
+                break;
+            case UserActionType.Search:
+                li = this.buildSearchEvent(action);
+                break;
+            case UserActionType.PageView:
+                li = this.buildViewEvent(action);
+                break;
+            case UserActionType.TicketCreated:
+                li = this.buildTicketCreated(action);
+                break;
+            default:
+            case UserActionType.Custom:
+                li = this.buildCustomEvent(action);
+                break;
+        }
         return li;
     }
```

Comment:
When having a `switch/case` with very similar cases, I tend to prefer using a map like this.

```suggestion
    private buildActionListItem(action: UserAction): HTMLLIElement {
      const defaultBuilder = this.buildCustomEvent;
      const builders = {
        [UserActionType.Click]: this.buildClickEvent,
        [UserActionType.Search]: this.buildSearchEvent,
        [UserActionType.PageView]: this.buildViewEvent,
        [UserActionType.TicketCreated]: this.buildTicketCreated,
        [UserActionType.Custom]: this.buildCustomEvent,
      };
      
      const builder = builders[action.type] ?? defaultBuilder;
      return builder(action);
    }
```

Reply:
I love it! This was code coming from the ancient implementation, but I prefer your way for sure!